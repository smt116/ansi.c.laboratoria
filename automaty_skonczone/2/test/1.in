/*
 * Maciej MaÅ‚ecki
 * smt116@gmail.com
 */
#include <stdio.h>
#include "lib.h"

void pobieranie_tekstu(char *prompt) {
    printf(prompt);
	// cos tu nietak chyba
    char znak;
    znak = getchar();

    while (znak != EOF) {
	if (znak == '/') {
	    usun_komentarze(&znak);
	} 
	else putchar(znak);
/* hmmm ciekawy jest *
ten komentarz *
aczkolwiek /* czy ja wiem.... */
	znak = getchar();
    }
}
void usun_komentarze(char *znak) {
    char znak1, znak2;
    znak1 = *znak;
    znak2 = getchar();

    if(znak1 == '/' && znak2 == '/') {
	    while(znak2=='\n'){
	    znak2 = getchar();}
    } else {
	    if(znak2 == '*') {
		    while(!(znak1=='*' && znak2=='/')){
			    znak1 = znak2;
			    znak2 = getchar();
		    }
	    } else {
		    putchar(znak1);
	    }
    }
}
