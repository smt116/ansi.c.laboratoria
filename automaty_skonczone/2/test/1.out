Wpisz kod zrodlowy (ctrl+d konczy wpisywanie)

#include <stdio.h>
#include "lib.h"

void pobieranie_tekstu(char *prompt) {
    printf(prompt);
	    char znak;
    znak = getchar();

    while (znak != EOF) {
	if (znak == '/) {
	    usun_komentarze(&znak);
	} 
	else putchar(znak);

	znak = getchar();
    }
}
void usun_komentarze(char *znak) {
    char znak1, znak2;
    znak1 = *znak;
    znak2 = getchar();

    if(znak1 == '/ && znak2 == '/) {
	    while(znak2=='\n'){
	    znak2 = getchar();}
    } else {
	    if(znak2 == '*') {
		    while(!(znak1=='*' && znak2=='/)){
			    znak1 = znak2;
			    znak2 = getchar();
		    }
	    } else {
		    putchar(znak1);
	    }
    }
}
